File: app/helpers/liquid_helper.rb
# Every single one of these to be reimplemented in JS in /app/assets/javascripts/liquid_helpers
module LiquidHelper
  def focusedImage(slug, metadata, aspect_ratio, *opts)
    opts = Hash[*opts]
    opts.merge({'fmt': 'pjpg', 'auto': 'compress'})
    image = FocusedImage.new(slug, metadata)
    aspect_ratio = aspect_ratio.split("x").map { |i| i.to_i}
    "//" + QtConfig.get.cdn_image + "/" + image.path(aspect_ratio, opts)
  end
end

File: app/assets/javascripts/liquid_helper.js
var quintype = require("quintype-js");
var _ = require("lodash");

Liquid.registerFilters({
  focusedImage: function(slug, metadata, aspect_ratio) {
    var image = new quintype.FocusedImage(slug, metadata);
    var aspectRatio = _.chain(aspect_ratio).split("x").map(function(x) { return parseInt(x); }).value();
    var opts = _.chain(arguments).drop(3).chunk(2).fromPairs().value();
    opts.fmt = 'pjpg'
    opts.auto = 'compress'
    return "//" + qtConfig.cdnImage + "/" + image.path(aspectRatio, opts);
  },

  partial_from_story_element: function(element) {
    var mapping = window.qtConfig.storyElementMapping;
    return mapping[element['subtype']] || mapping[element['type']]
  },

  base64_to_string: function(string) {
    return atob(string);
  }
});
